### PR TITLE
Add keycloak

### PR DESCRIPTION
**Pull request by drGrove**

### CODE REVIEW

File: keycloak/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - deployment.yaml
+  - service.yaml
+  - service-account.yaml
+  - ingress.yaml
+configMapGenerator:
+  - name: keycloak-config
+    literals:
+      - KC_HOSTNAME_URL=https://keycloak.hashbang.sh
+      - KC_DB=postgres
+      - KC_DB_URL_HOST=userdb-attempt-too-do-user-989073-0.db.ondigitalocean.com
```
Comment (daurnimator) [inline keycloak/kustomization.yaml:15]: I don't love that we're reusing this DB

File: keycloak/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - KC_DB_URL_PORT=25060
+      - KC_DB_URL_DATABASE=keycloak
+      - KC_DB_URL_PROPERTIES=?sslmode=require
+      - KC_DB_USERNAME=keycloak
+      - KEYCLOAK_ADMIN=admin
+generators:
+  - secret-generator.yaml
+images:
+  - name: quay.io/keycloak/keycloak
+    newTag: 23.0.2
```
Comment (daurnimator) [inline keycloak/kustomization.yaml:25]: needs digest

File: keycloak/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  - ingress.yaml
+configMapGenerator:
+  - name: keycloak-config
+    literals:
+      - KC_HOSTNAME_URL=https://keycloak.hashbang.sh
+      - KC_DB=postgres
+      - KC_DB_URL_HOST=userdb-attempt-too-do-user-989073-0.db.ondigitalocean.com
+      - KC_DB_URL_PORT=25060
+      - KC_DB_URL_DATABASE=keycloak
+      - KC_DB_URL_PROPERTIES=?sslmode=require
```
Comment (daurnimator) [inline keycloak/kustomization.yaml:18]: huh. this `?` is odd and barely undocumented

Comment (drGrove) [reply keycloak/kustomization.yaml:18]: It appends without the `?` unless it's provided

File: keycloak/kustomization.yaml (hunk 1 above)
Comment (drGrove) [reply keycloak/kustomization.yaml:15]: We're using this every service we currently have. I agree it's not ideal but DBs aren't cheap.

Comment (RyanSquared) [reply keycloak/kustomization.yaml:15]: > I don't love that we're reusing this DB

We are separating it into a separate database on the instance, though. I think that is fine.

File: keycloak/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                port:
+                  name: http
+          - path: /robots.txt
+            pathType: Exact
+            backend:
+              service:
+                name: keycloak
+                port:
+                  name: http
+
```
Comment (daurnimator) [inline keycloak/ingress.yaml:53]: empty newline

**Review by daurnimator — APPROVED** [review]

**Pull request merged**